conn.go

```diff
@@ -471,6 +477,29 @@ func (c *Conn) Send(destination, contentType string, body []byte, opts ...func(*
 	return nil
 }
 
+func readReceiptWithTimeout(request writeRequest, timeout time.Duration) error {
+	handle := func(response *frame.Frame) error {
+		if response.Command != frame.RECEIPT {
+			return newError(response)
+		}
+		return nil
+	}
+
+	if timeout <= 0 {
+		response := <-request.C
+		return handle(response)
+	}
+
+	timer := time.NewTimer(timeout)
+	select {
+	case <-timer.C:
+		return ErrMsgReceiptTimeout
+	case response := <-request.C:
+		timer.Stop()
+		return handle(response)
+	}
```

Comment:
Can I suggest simplifying this function by always using the timeout channel?

```suggestion
	var ch <-chan time.Time
	if timeout > 0 {
		ch = time.After(timeout)
	}
	select {
	case <-ch:
		return ErrMsgReceiptTimeout
	case response := <-request.C:
		if response.Command != frame.RECEIPT {
			return newError(response)
		}
		return nil
	}
```

Reply:
This was to follow the established pattern in the method below, but the approach here is undoubtedly cleaner. Will update